core/src/collections.rs

```diff
@@ -11,48 +11,55 @@ use crate::{
 use bson::{doc, Bson};
 use definitions::Nullability;
 use futures::future;
-use lazy_static::lazy_static;
 use mongodb::{options::ListDatabasesOptions, results::CollectionType};
+use once_cell::sync::OnceCell;
 use regex::Regex;
 
-lazy_static! {
-    static ref COLLECTIONS_METADATA: Vec<MongoColMetadata> = vec![
+static COLLECTIONS_METADATA: OnceCell<Vec<MongoColMetadata>> = OnceCell::new();
+
+fn init_collections_metadata(max_string_length: Option<u16>) -> Vec<MongoColMetadata> {
```

Comment:
Generally, the "static" metadata now requires runtime information (whether or not the string length limit is imposed). So switched these all from `lazy_static`s to `OnceCell`s which are set lazily -- exactly once -- and then retrieved over and over again without recomputing.